cuqi/solver/_solver.py

```diff
@@ -650,8 +650,124 @@ def solve(self):
                 x_new = x_new + ((k-1)/(k+2))*(x_new - x_old)
               
             x = x_new.copy()
+
+class ADMM(object):
+    """Alternating Direction Method of Multipliers for solving regularized linear least squares problems of the form:
+    Minimize ||Ax-b||^2 + sum_i f_i(L_i x).
+
+    Reference:
+    Boyd et al. "Distributed optimization and statistical learning via the alternating direction method of multipliers."Foundations and Trends® in Machine learning, 2011.
+
     
+    Parameters
+    ----------
+    A : ndarray or callable f(x,*args).
+    b : ndarray.
+    penalties : List of tuples (callable proximal operator of f_i, linear operator L_i).
+    x0 : ndarray. Initial guess.
+    tradeoff : Trade-off between linear least squares and regularization term in the solver iterates.
```

Comment:
Is "tradeoff" the term used in the Boyd reference? If not can you add a a bit more explanation. If it is a term used, can you add to see the reference paper?

Reply:
The original name used for this in the reference is "penalty parameter", however, as I refer to the list of regularization/constraints as penalties, I thought this would be confusing.  Alternatively, the character "ρ" is used in the equations of the reference, hence why it is called rho inside the class. I thought that "tradeoff" would better describe the interpretation of the variable.

Any suggestions on how/what to modify?

Reply:
```suggestion
    tradeoff : Trade-off between linear least squares and regularization term in the solver iterates. Denoted as "rho" in [1].
```
Then add [1] as your reference.